### PR TITLE
DotNet: add LLama Connector and its example

### PR DESCRIPTION
**Pull request by mifmasterz**
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by mifmasterz** [comment]
> @mifmasterz please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Comment by mifmasterz** [comment]
fixing all issues

File: dotnet/src/Connectors/Connectors.AI.PaLM/Connectors.AI.PaLM.csproj [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (shawncal) [inline dotnet/src/Connectors/Connectors.AI.PaLM/Connectors.AI.PaLM.csproj:1]: Revert the PaLM stuff? I see there's another PR for that...

File: dotnet/src/Connectors/Connectors.AI.LLama/Directory.Packages.props [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+<Project>
+  <ItemGroup>
+    <PackageReference Include="LLamaSharp" Version="0.4.0" />
```
Comment (shawncal) [inline dotnet/src/Connectors/Connectors.AI.LLama/Directory.Packages.props:4]: Please move these to the common Directory.Packages.props under /dotnet

File: dotnet/src/Connectors/Connectors.AI.LLama/Connectors.AI.LLama.csproj [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+  </PropertyGroup>
+  <!-- IMPORT NUGET PACKAGE SHARED PROPERTIES -->
```
Comment (shawncal) [inline dotnet/src/Connectors/Connectors.AI.LLama/Connectors.AI.LLama.csproj:6]:
```suggestion
  <!-- IMPORT NUGET PACKAGE SHARED PROPERTIES -->
  <Import Project="$(RepoRoot)/dotnet/nuget/nuget-package.props" />
```

File: dotnet/src/Connectors/Connectors.AI.LLama/Connectors.AI.LLama.csproj [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (shawncal) [inline dotnet/src/Connectors/Connectors.AI.LLama/Connectors.AI.LLama.csproj:4]:
```suggestion
    <!-- THIS PROPERTY GROUP MUST COME FIRST -->
    <AssemblyName>Microsoft.SemanticKernel.Connectors.AI.LLama</AssemblyName>
    <RootNamespace>$(AssemblyName)</RootNamespace>
    <TargetFramework>netstandard2.0</TargetFramework>
```

**Review by shawncal — CHANGES_REQUESTED** [review]

File: dotnet/src/Connectors/Connectors.AI.PaLM/Connectors.AI.PaLM.csproj (hunk 1 above)
Comment (mifmasterz) [reply dotnet/src/Connectors/Connectors.AI.PaLM/Connectors.AI.PaLM.csproj:1]: my mistake, that's my another PR.. sorry

File: dotnet/src/Connectors/Connectors.AI.LLama/Directory.Packages.props (hunk 2 above)
Comment (mifmasterz) [reply dotnet/src/Connectors/Connectors.AI.LLama/Directory.Packages.props:4]: moved ok

**Comment by davearlin** [comment]
@mifmasterz  I just want to say that I'm really excited for this particular PR to get in (and the PaLM one too).  If I could be of any assistance, I'd love to help where I can although I'm not a cpp guy.   😄

**Pull request closed without merging**